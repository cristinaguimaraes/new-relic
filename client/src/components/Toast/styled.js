import styled from 'styled-components';

export const TextWrapper = styled.div`
  width: 100%;
  box-shadow: 0 2px 12px 4px rgba(0, 0, 0, 0.19);
  border-radius: 6px;
  color: white;
  justify-content: flex-start;
  flex-direction: row;
  padding: 16px;
  display: flex;
  align-items: center;
  background-color: green;
`;

export const Text = styled.span`
  margin-left: 8px
`;

export const IconWrapper = styled.div`
  display: flex;
  align-items: center;
  margin-left: auto;
`;
