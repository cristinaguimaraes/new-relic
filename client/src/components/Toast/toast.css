.custom-background {
  background-color: transparent;
  box-shadow: none;
  height: 100px;
  padding: 0;
}

@media only screen and (min-width: 600px) {
  .custom-background {
    width: 100vw;
    left: 0;
    margin: 0;
    height: 100px;
  }
}

.Toastify__toast-container {
  padding: 0;
}

.Toastify__toast-container--top-center {
  top: 96px;
  margin-left: auto;
  margin-right: auto;
  width: 100vw;
  left: 0;
}

@media only screen and (min-width: 600px) {
  .Toastify__toast-container--top-center {
    top: 45px;
  }
}

@keyframes fadeInDown {
  from {
    opacity: 0;
    transform: translate3d(0, -100%, 0);
  }

  to {
    opacity: 1;
    transform: translate3d(0, 0, 0);
  }
}

.custom-fadeInDown {
  animation-name: fadeInDown;
}

@keyframes fadeOutUp {
  from {
    opacity: 1;
  }

  to {
    opacity: 0;
    transform: translate3d(0, -100%, 0);
  }
}

.custom-fadeOutUp {
  animation-name: fadeOutUp;
}
