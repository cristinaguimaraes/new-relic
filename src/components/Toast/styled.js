import styled from "styled-components";

export const TextWrapper = styled.div`
  width: 50%;
  box-shadow: 0 2px 12px 4px rgba(0, 0, 0, 0.19);
  border-radius: 6px !important;
  justify-content: flex-start;
  flex-direction: row;
  padding: 16px;
  display: flex; 
  align-items: center;
  background-color: ${props => (props.type === 'failure' ? 'red' : 'green')};
  color: white;
`;

export const Text = styled.span`
  margin-left: 8px;
`;
