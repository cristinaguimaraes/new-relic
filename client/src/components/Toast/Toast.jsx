import React, { useEffect } from 'react';
import { bool, func, string } from 'prop-types';
import { cssTransition, toast, ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import './toast.css';


import { TextWrapper, Text} from './styled';
import { Icon } from '../AppsByHostHeader/styled';

const Fade = cssTransition({
  enter: 'custom-fadeInDown',
  exit: 'custom-fadeOutUp',
  duration: 500
});

const Toast = ({ showToast, messageKey, onClose  }) => {

  useEffect(() => {
    if (showToast) {
      toast(<Message />, {
        onClose,
        className: 'custom-background',
        containerId: 'custom-toast-container',
      });
    }
  }, [showToast]);
  
  
  const Message = () => (
  
          <TextWrapper>
            <Icon className={'material-icons'} >
              'check_circle'
            </Icon>
            <Text>{messageKey}</Text>
            <Icon className={'material-icons'} onClick={onClose}>
              'close'
            </Icon>
          </TextWrapper>
  
  );

  return (
    <ToastContainer
      containerId={'custom-toast-container'}
      autoClose={2500}
      draggable
      position={toast.POSITION.TOP_CENTER}
      closeButton={false}
      newestOnTop
      closeOnClick
      pauseOnHover={false}
      hideProgressBar
      transition={Fade}
      type={'success'}
    />
  );
};

Toast.propTypes = {
  showToast: bool,
  messageKey: string,
  onClose: func
};

export { Toast };
