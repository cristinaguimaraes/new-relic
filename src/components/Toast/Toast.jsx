import React, { useEffect } from "react";
import { bool, func, oneOf, string } from "prop-types";
import { cssTransition, toast, ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import "./toast.css";
import { TextWrapper, Text } from "./styled";

const Fade = cssTransition({
  enter: "custom-fadeInDown",
  exit: "custom-fadeOutUp",
  duration: 500
});

const Toast = ({ showToast, messageKey, onClose, type }) => {
  useEffect(() => {
    if (showToast) {
      toast(<Message />, {
        onClose,
        className: "custom-background",
        containerId: "custom-toast-container"
      });
    }
    // eslint-disable-next-line
  }, [showToast]);

  const Message = () => (
    <TextWrapper type={type}>
      <Text>{messageKey}</Text>
    </TextWrapper>
  );

  return (
    <ToastContainer
      containerId={"custom-toast-container"}
      autoClose={2500}
      draggable
      position={toast.POSITION.TOP_RIGHT}
      closeButton={false}
      newestOnTop
      closeOnClick
      pauseOnHover={false}
      hideProgressBar
      transition={Fade}
      type={"success"}
    />
  );
};

Toast.propTypes = {
  showToast: bool,
  messageKey: string,
  onClose: func,
  type: oneOf(["success", "failure"])
};

export { Toast };
