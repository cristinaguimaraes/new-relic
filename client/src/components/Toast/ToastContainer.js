import { connect } from 'react-redux';
import { Toast } from './Toast';
import { onHideToastMessage } from '../../redux/actions/creators';

const mapStateToProps = state => ({
  showToast: state.toast.show,
  messageKey: state.toast.messageKey
});

const mapDispatchToProps = {
  onClose: onHideToastMessage
};


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Toast);
