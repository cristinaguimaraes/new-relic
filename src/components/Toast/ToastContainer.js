import { connect } from "react-redux";
import { Toast } from "./Toast";
import { onHideToastMessage } from "redux/actions/creators";

const mapStateToProps = state => ({
  showToast: state.toast.show,
  messageKey: state.toast.messageKey,
  type: state.toast.type
});

const mapDispatchToProps = {
  onClose: onHideToastMessage
};

export default connect(mapStateToProps, mapDispatchToProps)(Toast);
