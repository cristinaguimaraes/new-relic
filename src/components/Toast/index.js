import ToastContainer from "./ToastContainer";
export { ToastContainer as Toast };
