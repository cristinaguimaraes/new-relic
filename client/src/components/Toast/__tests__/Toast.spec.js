import { Toast } from '../Toast';
import { shallow } from 'enzyme';
import React from 'react';

describe('Toast component', () => {
  test('should render properly', () => {
    const fn = () => {};
    const wrapper = shallow(<Toast showToast={false} onHideToast={fn} onClose={fn} />);
    expect(wrapper).toMatchSnapshot();

    const wrapper1 = shallow(<Toast messageKey={'details'} showToast={true} onHideToast={fn} onClose={fn} />);
    expect(wrapper1).toMatchSnapshot();
  });
});
