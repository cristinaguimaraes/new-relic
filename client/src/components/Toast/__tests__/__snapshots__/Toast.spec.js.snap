// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component should render properly 1`] = `
<ToastContainer
  autoClose={2500}
  bodyClassName={null}
  className={null}
  closeButton={false}
  closeOnClick={true}
  containerId="custom-toast-container"
  draggable={true}
  draggablePercent={80}
  hideProgressBar={true}
  newestOnTop={true}
  pauseOnFocusLoss={true}
  pauseOnHover={false}
  position="top-center"
  progressClassName={null}
  progressStyle={null}
  role="alert"
  rtl={false}
  style={
    Object {
      "marginTop": 0,
    }
  }
  toastClassName={null}
  transition={[Function]}
/>
`;

exports[`Toast component should render properly 2`] = `
<ToastContainer
  autoClose={2500}
  bodyClassName={null}
  className={null}
  closeButton={false}
  closeOnClick={true}
  containerId="custom-toast-container"
  draggable={true}
  draggablePercent={80}
  hideProgressBar={true}
  newestOnTop={true}
  pauseOnFocusLoss={true}
  pauseOnHover={false}
  position="top-center"
  progressClassName={null}
  progressStyle={null}
  role="alert"
  rtl={false}
  style={
    Object {
      "marginTop": 0,
    }
  }
  toastClassName={null}
  transition={[Function]}
/>
`;
